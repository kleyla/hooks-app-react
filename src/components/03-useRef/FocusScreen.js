import React, { useRef } from "react";
import "../02-useEffect/effects.css";

export const FocusScreen = () => {
  const inputRef = useRef();

  const handleClick = () => {
    // document.querySelector("input").select();
    inputRef.current.select();
    console.log(inputRef);
  };

  return (
    <div>
      <h1>Focus Screen</h1>
      <hr />
      <input
        type="text"
        name=""
        className="form-control"
        placeholder="Su nombre"
        ref={inputRef}
      />
      <button className="btn brn-outline-primary mt-5" onClick={handleClick}>
        Focus
      </button>
    </div>
  );
};
