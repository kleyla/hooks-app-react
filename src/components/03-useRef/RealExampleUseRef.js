import React, { useState } from "react";
import "../02-useEffect/effects.css";
import { MultipleCustomHooks } from "../examples/MultipleCustomHooks";

export const RealExampleUseRef = () => {
  const [show, setShow] = useState(false);

  return (
    <div>
      <h1>Real Example useRef</h1>
      <hr />

      {show && <MultipleCustomHooks />}

      <button className="btn btn-primary mt-4" onClick={() => setShow(!show)}>
        Show/hide
      </button>
    </div>
  );
};
