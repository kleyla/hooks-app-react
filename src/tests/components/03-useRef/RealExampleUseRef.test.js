import { shallow } from "enzyme";
import { RealExampleUseRef } from "../../../components/03-useRef/RealExampleUseRef";

describe("Pruebas en RealExampleUseRef", () => {
  const wrapper = shallow(<RealExampleUseRef />);

  test("Debe mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(false);
  });

  test("Debe de mostrar el componente MultipleCustomHooks", () => {
    wrapper.find("button").simulate("click");
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(true);
  });
});
