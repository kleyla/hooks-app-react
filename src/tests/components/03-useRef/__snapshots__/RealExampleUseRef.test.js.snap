// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en RealExampleUseRef Debe mostrarse correctamente 1`] = `
<div>
  <h1>
    Real Example useRef
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-4"
    onClick={[Function]}
  >
    Show/hide
  </button>
</div>
`;
